infrastructure/zk/src/spellcheck.ts

```diff
@@ -1,28 +1,34 @@
 import { Command } from 'commander';
 import * as utils from './utils';
 
-export async function runSpellCheck(pattern: string, config: string, useCargo: boolean, useCSpell: boolean) {
+export async function runSpellCheck(pattern: string, useCargo: boolean, useCSpell: boolean) {
     const cSpellCommand = `cspell ${pattern} --config=./spellcheck/cspell.json`;
     const cargoCommand = `cargo spellcheck --cfg=./spellcheck/era.cfg`;
 
     try {
+        let results = [];
         if (!useCargo && !useCSpell) {
-            await Promise.all([utils.spawn(cSpellCommand), utils.spawn(cargoCommand)]);
+            results = await Promise.all([utils.spawn(cSpellCommand), utils.spawn(cargoCommand)]);
         } else {
             // Run cspell if specified
-            // zk spellcheck --use-cspell
             if (useCSpell) {
-                await utils.spawn(cSpellCommand);
+                results.push(await utils.spawn(cSpellCommand));
             }
 
             // Run cargo spellcheck if specified
-            // zk spellcheck --use-cargo
             if (useCargo) {
-                await utils.spawn(cargoCommand);
+                results.push(await utils.spawn(cargoCommand));
             }
         }
+
+        // Check results and exit with error code if any command failed
+        if (results.some((code) => code !== 0)) {
+            console.error('Spell check failed');
+            process.exit(1);
+        }
     } catch (error) {
         console.error('Error occurred during spell checking:', error);
+        process.exit(1);
     }
```

Comment:
do you even need this try catch? 
i don't see much value here

Reply:
Yes its needed to correctly handle rejected promises and properly displays the error.

Reply:
`Error occurred during spell checking: Child process exited with code 1` 

Are you sure?
If it's really necessary I don't mind